iceoryx_utils/include/iceoryx_utils/cxx/vector.hpp

```diff
@@ -69,75 +74,62 @@ class vector
 
     /// @brief returns an iterator to the first element of the vector,
     ///         if the vector is empty it returns the same iterator as
-    ///         end (the first iterator which is outside of the vector)
-    iterator begin();
+    ///         end (nullptr)
```

Comment:
please don't state that it's a `nullptr`. It's just an implementian detail and users should always compare to `end()`. If they start comparing to a `nullptr` we are locked into this implementation.

Reply:
Clear. The implementation no longer returns a nullptr, but I forgot the comment...